kubernetes helm chart ingress annotations mis-typed

**Describe the bug**

The helm chart's `values.yaml`  defines `ingress.annotations:` without the `{}` after it. This causes Helm to expect a list-array instead of a key-value array.  I couldn't find a way to make that work.

Editing `values.yaml` to use `{}` for the ingress annotations, like all the other types' annotations entries, let my ingress annotations work.

**Error Message**
If applicable add error message to help explain your problem or console output enclose in triple back-ticks eg,

```
coalesce.go:199: warning: destination for annotations is a table. Ignoring non-table value <nil>
```

**To Reproduce**

Try the helm chart (current master branch, `90af973c3d`) with some ingress annotations.

**Expected behavior**
Able to provide annotations to Helm which propagate into the Ingress object.

**Environment (please complete the following information):**
Helm `v3.1.2`, Kubernetes server version `v1.17.4+k3s1`

**Additional context**
It's a one-line fix, adding two characters, not opening a PR for that.  :laughing:

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -44,22 +44,3 @@ jobs:
         id: chart-install
         run: ct install --remote origin --all
         if: steps.list-changed.outputs.changed == 'true'
-
-  release:
-    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'
-    name: Publish Helm Chart
-    runs-on: ubuntu-latest
-    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
-    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
-    permissions:
-      contents: write
-    steps:
-      - name: Configure Git
-        run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
-
-      - name: Run chart-releaser
-        uses: helm/chart-releaser-action@v1.5.0
-        env:
-          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -2,7 +2,7 @@ name: Helm Charts
 on:
   push:
     branches:
-      - github-actions
+      - main
 
   pull_request:
       types: [opened, reopened, synchronize]

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -38,12 +38,12 @@ jobs:
 
       - name: Create kind cluster
         uses: helm/kind-action@v1.4.0
-#        if: steps.list-changed.outputs.changed == 'true'
+        if: steps.list-changed.outputs.changed == 'true'
 
       - name: Run chart-testing (install)
         id: chart-install
-        run: |
-          ct install --remote origin --all && echo "::set-output name=success::true"
+        run: ct install --remote origin --all
+        if: steps.list-changed.outputs.changed == 'true'
 
 #  release:
 #    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -38,7 +38,7 @@ jobs:
 
       - name: Create kind cluster
         uses: helm/kind-action@v1.4.0
-        if: steps.list-changed.outputs.changed == 'true'
+#        if: steps.list-changed.outputs.changed == 'true'
 
       - name: Run chart-testing (install)
         id: chart-install

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -10,22 +10,14 @@ on:
           - "charts/**"
 
 jobs:
-  release:
-    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
-    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
-    permissions:
-      contents: write
+  test:
+    name: Test Helm Chart
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
-
-      - name: Configure Git
-        run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
 
       - name: Install Helm
         uses: azure/setup-helm@v3
@@ -38,7 +30,7 @@ jobs:
         run: |
           changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
           if [[ -n "$changed" ]]; then
-            echo "::set-output name=changed::true"
+            echo "changed=true" >> $GITHUB_OUTPUT
           fi
 
       - name: Run chart-testing (lint)
@@ -53,8 +45,21 @@ jobs:
         run: |
           ct install --remote origin --chart-dirs=${CHARTS_DIR} --all && echo "::set-output name=success::true"
 
+#  release:
+#    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'
+#    name: Publish Helm Chart
+#    runs-on: ubuntu-latest
+#    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
+#    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
+#    permissions:
+#      contents: write
+#    steps:
+#      - name: Configure Git
+#        run: |
+#          git config user.name "$GITHUB_ACTOR"
+#          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+#
 #      - name: Run chart-releaser
 #        uses: helm/chart-releaser-action@v1.5.0
 #        env:
 #          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-#        if: steps.chart-install.output.success == 'true'

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -23,7 +23,7 @@ jobs:
         uses: azure/setup-helm@v3
 
       - name: Set up chart-testing
-        uses: helm/chart-testing-action@v2.3.1
+        uses: helm/chart-testing-action@v3.7.1
 
       - name: Run chart-testing (list-changed)
         id: list-changed
@@ -37,7 +37,7 @@ jobs:
         run: ct lint --target-branch ${{ github.event.repository.default_branch }}
 
       - name: Create kind cluster
-        uses: helm/kind-action@v1.4.0
+        uses: helm/kind-action@v1.5.0
         if: steps.list-changed.outputs.changed == 'true'
 
       - name: Run chart-testing (install)

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -43,7 +43,7 @@ jobs:
       - name: Run chart-testing (install)
         id: chart-install
         run: |
-          ct install --remote origin --chart-dirs=${CHARTS_DIR} --all && echo "::set-output name=success::true"
+          ct install --remote origin --all && echo "::set-output name=success::true"
 
 #  release:
 #    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -1,0 +1,60 @@
+name: Helm Charts
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+      types: [opened, reopened, synchronize]
+      paths:
+          - "charts/**"
+
+jobs:
+  release:
+    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
+    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.3.1
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch }})
+          if [[ -n "$changed" ]]; then
+            echo "::set-output name=changed::true"
+          fi
+
+      - name: Run chart-testing (lint)
+        run: ct lint --target-branch ${{ github.event.repository.default_branch }}
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.4.0
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Run chart-testing (install)
+        id: chart-install
+        run: |
+          ct install --remote origin --chart-dirs=${CHARTS_DIR} --all && echo "::set-output name=success::true"
+
+#      - name: Run chart-releaser
+#        uses: helm/chart-releaser-action@v1.5.0
+#        env:
+#          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+#        if: steps.chart-install.output.success == 'true'

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -23,7 +23,7 @@ jobs:
         uses: azure/setup-helm@v3
 
       - name: Set up chart-testing
-        uses: helm/chart-testing-action@v3.7.1
+        uses: helm/chart-testing-action@v2.3.1
 
       - name: Run chart-testing (list-changed)
         id: list-changed

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -2,7 +2,7 @@ name: Helm Charts
 on:
   push:
     branches:
-      - main
+      - github-actions
 
   pull_request:
       types: [opened, reopened, synchronize]

--- a/.github/workflows/helm.yaml
+++ b/.github/workflows/helm.yaml
@@ -45,21 +45,21 @@ jobs:
         run: ct install --remote origin --all
         if: steps.list-changed.outputs.changed == 'true'
 
-#  release:
-#    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'
-#    name: Publish Helm Chart
-#    runs-on: ubuntu-latest
-#    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
-#    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
-#    permissions:
-#      contents: write
-#    steps:
-#      - name: Configure Git
-#        run: |
-#          git config user.name "$GITHUB_ACTOR"
-#          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
-#
-#      - name: Run chart-releaser
-#        uses: helm/chart-releaser-action@v1.5.0
-#        env:
-#          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+  release:
+    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/main'
+    name: Publish Helm Chart
+    runs-on: ubuntu-latest
+    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
+    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
+    permissions:
+      contents: write
+    steps:
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.5.0
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -1,0 +1,24 @@
+name: Helm Charts
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  release:
+    name: Publish Helm Chart
+    runs-on: ubuntu-latest
+    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
+    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
+    permissions:
+      contents: write
+    steps:
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.5.0
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"

--- a/charts/athens-proxy/Chart.yaml
+++ b/charts/athens-proxy/Chart.yaml
@@ -1,15 +1,17 @@
 apiVersion: v1
 name: athens-proxy
 version: 0.5.5
-appVersion: 0.11.1
+appVersion: 0.11.0
 description: The proxy server for Go modules
 icon: https://raw.githubusercontent.com/gomods/athens/main/docs/static/banner.png
 keywords:
 - Golang
 - Package Management
 - Goproxy
+- Athens
 home: https://github.com/gomods/athens
 sources:
+  - https://github.com/SickHub/athens-proxy
 maintainers:
 - name: rimusz
   email: rmocius@gmail.com

--- a/charts/athens-proxy/Chart.yaml
+++ b/charts/athens-proxy/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: athens-proxy
-version: 0.5.4
+version: 0.5.5
 appVersion: 0.11.1
 description: The proxy server for Go modules
 icon: https://raw.githubusercontent.com/gomods/athens/main/docs/static/banner.png

--- a/charts/athens-proxy/Chart.yaml
+++ b/charts/athens-proxy/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: athens-proxy
 version: 0.5.5
-appVersion: 0.11.0
+appVersion: v0.11.0
 description: The proxy server for Go modules
 icon: https://raw.githubusercontent.com/gomods/athens/main/docs/static/banner.png
 keywords:

--- a/charts/athens-proxy/templates/_helpers.tpl
+++ b/charts/athens-proxy/templates/_helpers.tpl
@@ -14,6 +14,3 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this 
 {{- end -}}
 {{- end -}}
 {{- end -}}
-{{- define "readinessPath" -}}
-{{- if contains "v0.2.0" .Values.image.tag -}}/{{- else -}}/readyz{{- end -}}
-{{- end -}}

--- a/charts/athens-proxy/templates/deployment.yaml
+++ b/charts/athens-proxy/templates/deployment.yaml
@@ -58,7 +58,7 @@ spec:
       {{- end }}
       containers:
       - name: {{ template "fullname" . }}
-        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
         imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         livenessProbe:
           failureThreshold: {{ .Values.livenessProbe.failureThreshold }}

--- a/charts/athens-proxy/templates/deployment.yaml
+++ b/charts/athens-proxy/templates/deployment.yaml
@@ -70,7 +70,7 @@ spec:
           timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
         readinessProbe:
           httpGet:
-            path: "{{ template "readinessPath" . }}"
+            path: "/readyz"
             port: 3000
         env:
         - name: ATHENS_GOGET_WORKERS

--- a/charts/athens-proxy/templates/test/test-connection.yaml
+++ b/charts/athens-proxy/templates/test/test-connection.yaml
@@ -1,0 +1,15 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "fullname" . }}-test-connection"
+  labels:
+    app: {{ template "fullname" . }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args: ['{{ include "fullname" . }}:{{ .Values.service.servicePort }}']
+  restartPolicy: Never

--- a/charts/athens-proxy/values.yaml
+++ b/charts/athens-proxy/values.yaml
@@ -128,12 +128,12 @@ upstreamProxy:
   url: "https://gocenter.io"
 
 jaeger:
+  enabled: false
   ## Type of service; valid values are "ClusterIP", "LoadBalancer", and "NodePort".
   type: ClusterIP
   image:
     repository: jaegertracing/all-in-one
     tag: latest
-  enabled: true
   # you must set this on the command line when you run 'helm install'
   # for example, you need to run 'helm install --set jaeger.url=myurl ...'
   url: "SET THIS ON THE COMMAND LINE"

--- a/charts/athens-proxy/values.yaml
+++ b/charts/athens-proxy/values.yaml
@@ -41,7 +41,7 @@ service:
 ingress:
   enabled: false
   # Provide key/value annotations
-  annotations:
+  annotations: {}
   className: ""
   # Provide an array of values for the ingress host mapping
   hosts:

--- a/charts/athens-proxy/values.yaml
+++ b/charts/athens-proxy/values.yaml
@@ -2,7 +2,7 @@ replicaCount: 1
 image:
   registry: docker.io
   repository: gomods/athens
-  tag: v0.11.0
+  # tag: v0.11.0
 
   ## Specify a imagePullPolicy
   ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'

--- a/charts/athens-proxy/values.yaml
+++ b/charts/athens-proxy/values.yaml
@@ -2,6 +2,7 @@ replicaCount: 1
 image:
   registry: docker.io
   repository: gomods/athens
+  # Override the chart appVersion and use a specific tag
   # tag: v0.11.0
 
   ## Specify a imagePullPolicy

